internal/resources/oncall/resource_integration.go

```diff
@@ -205,6 +215,131 @@ func ResourceIntegration() *schema.Resource {
 							},
 							MaxItems: 1,
 						},
+						"web": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "Templates for Web.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"title": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "Template for Alert title.",
+									},
+									"message": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "Template for Alert message.",
+									},
+									"image_url": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "Template for Alert image url.",
+									},
+								},
+							},
+							MaxItems: 1,
+						},
+						"telegram": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "Templates for Telegram.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"title": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "Template for Alert title.",
+									},
+									"message": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "Template for Alert message.",
+									},
+									"image_url": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "Template for Alert image url.",
+									},
+								},
+							},
+							MaxItems: 1,
+						},
```

Comment:
These blocks are all very similar, they could be generated by a single function
```suggestion
func onCallTemplate(name string, hasMessage, hasImage bool) *schema.Schema
...
						"telegram": onCallTemplate("Telegram", true, true),
```